src/lib.rs

```diff
@@ -0,0 +1,57 @@
+use std::fs;
+use std::io::Result;
+use std::path::{Path, PathBuf};
+
+pub struct Config {
+    path: PathBuf,
+    root: String,
+}
+
+impl Config {
+    pub fn new(root: impl Into<String>, path: impl Into<String>) -> Self {
+        Config {
+            path: Path::new(&path.into()).to_owned(),
```

Comment:
Why are we using &path.into()).to_owned() here?
I would assume it's because you tried to make Config accept &str?

Reply:
The idea is that by having `path` be `impl Into<String>` we can pass both `String` and `&str`, making the API more easy to use.
Howver, we want to store `path` not just as *any* random UTF-8 string, but as an actual * valid path*, thus the need to transform it. You can play around and you will quickly notice in order to store a `Path` it's more easily persisted in a struct as `PathBuf` or `&Path` .
I know it looks a bit complicated, I've tried to simplify it myself.